test/gtest/ucp/test_ucp_tag_mem_type.cc

```diff
@@ -0,0 +1,206 @@
+/**
+* Copyright (C) Mellanox Technologies Ltd. 2001-2019.  ALL RIGHTS RESERVED.
+* Copyright (C) UT-Battelle, LLC. 2015. ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#include "test_ucp_tag.h"
+#include <common/mem_buffer.h>
+
+#include "ucp_datatype.h"
+
+extern "C" {
+#include <ucp/core/ucp_ep.inl>
+#include <ucs/datastruct/queue.h>
+}
+
+#include <iostream>
+
+
+class test_ucp_tag_mem_type: public test_ucp_tag {
+public:
+    enum {
+            VARIANT_DEFAULT     = UCS_BIT(0),
+            VARIANT_GDR_OFF     = UCS_BIT(1),
+            VARIANT_TAG_OFFLOAD = UCS_BIT(2),
```

Comment:
btw, gdr_off and tag_offload together are buggy:)
added fix and simple test for that in #4499 
maybe worth adding another variant combining them